src/github_actions_exporter.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper class used to manage interactions with GitHub Actions Exporter."""
+
+from re import findall
+
+
+# There is only one method for this class.
+class GitHubActionsExporter:  # pylint: disable=too-few-public-methods
+    """This class handles interactions with GitHub Actions Exporter."""
+
+    def __init__(
+        self,
+        charm,
+    ):
+        """Construct."""
+        self._charm = charm
```

Comment:
The guidance is not to pass the entire `charm` here since it means that this module is coupled to the charm. The container can be passed for file system interactions. Perhaps take a look at the 12-factor charm and see what kind of patterns are established there?